Amazon.QLDB.Driver/transaction/AsyncTransactionExecutor.cs

```diff
@@ -52,6 +54,19 @@ public Task<IAsyncResult> Execute(string statement)
             return this.transaction.Execute(statement);
         }
 
+        /// <summary>
+        /// Execute the IQuery against QLDB and retrieve the result.
+        /// </summary>
+        public  Task<Amazon.QLDB.Driver.Generic.IAsyncResult<T>> Execute<T>(IQuery<T> query)
+        {
+            return this.transaction.Execute(query);
+        }
+
+        public IQuery<T> Query<T>(string statement, params object[] parameters)
```

Comment:
Given that you have to type the Query method, and that Execute is not the same as Execute<T>, why do we need the Query method at all? Why could we not use the presence (or absence) of the type as indication of using serialization, and do away with the QueryData object as an explicit reference. Seems much simpler.

Reply:
`tx.Query` allows the driver to get access to the desired de/serializer. It is not *required* to create an `IQuery` if you the developer want to create your `IQuery` some other way. I deliberately introduced `IQuery`so thay we could eventually support different types of queries instead of adding more and more methods to the `TransactionExecutor` interface.

Reply:
For example, adding the following method:

`public Task<Amazon.QLDB.Driver.Generic.IAsyncResult<T>> Execute<T>(string statement, params object[] parameters) {
  return this.transaction.Execute(new QueryData<T>(statement, parameters, serializer);
}
`